tests/enterprise-contract/contract.go

```diff
@@ -51,23 +85,274 @@ var _ = framework.EnterpriseContractSuiteDescribe("Enterprise Contract E2E tests
 		verifyECTaskBundle = cm.Data["verify_ec_task_bundle"]
 		Expect(verifyECTaskBundle).ToNot(BeEmpty())
 		GinkgoWriter.Printf("Using verify EC task bundle: %s\n", verifyECTaskBundle)
-		imageWithDigest = "quay.io/redhat-appstudio/ec-golden-image@sha256:4b318620a32349fd37827163c67b5ff6e503f05b3ca4dde066ee03bb34be9ae1"
 	})
-	Context("ec-cli command verification", func() {
-		BeforeEach(func() {
-			generator = tekton.VerifyEnterpriseContract{
-				TaskBundle:          verifyECTaskBundle,
-				Name:                "verify-enterprise-contract",
-				Namespace:           namespace,
-				PolicyConfiguration: "ec-policy",
-				PublicKey:           fmt.Sprintf("k8s://%s/%s", namespace, publicSecretName),
-				Strict:              false,
-				EffectiveTime:       "now",
-				IgnoreRekor:         true,
+
+	Context("infrastructure is running", Label("pipeline"), func() {
+		It("verifies if the chains controller is running", func() {
+			err := kubeClient.CommonController.WaitForPodSelector(kubeClient.CommonController.IsPodRunning, constants.TEKTON_CHAINS_NS, "app", "tekton-chains-controller", 60, 100)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("verifies the signing secret is present", func() {
+			timeout := time.Minute * 5
+			interval := time.Second * 1
+
+			Eventually(func() bool {
+				config, err := kubeClient.CommonController.GetSecret(constants.TEKTON_CHAINS_NS, constants.TEKTON_CHAINS_SIGNING_SECRETS_NAME)
+				Expect(err).NotTo(HaveOccurred())
+
+				_, private := config.Data["cosign.key"]
+				_, public := config.Data["cosign.pub"]
+				_, password := config.Data["cosign.password"]
+
+				return private && public && password
+			}, timeout, interval).Should(BeTrue(), fmt.Sprintf("timed out when waiting for Tekton Chains signing secret %q to be present in %q namespace", constants.TEKTON_CHAINS_SIGNING_SECRETS_NAME, constants.TEKTON_CHAINS_NS))
+		})
+	})
+
+	Context("test creating and signing an image and task", Label("pipeline"), func() {
+		// Make the PipelineRun name and namespace predictable. For convenience, the name of the
+		// PipelineRun that builds an image, is the same as the repository where the image is
+		// pushed to.
+		var buildPipelineRunName, image, imageWithDigest string
```

Comment:
Re-declaring `imageWithDigest` here. This might be a good thing as it is now being declared closer to where it is being used. Remove the top-level `imageWithDigest` declaration.